### ISSUE TITLE
Tennagraph donation in limbo

### ISSUE THREAD

**Issue by GriffGreen** [Giveth/giveth-dapp#561]
I delegated a bunch of ETH to https://beta.giveth.io/campaigns/5b51bdbdf8ba4732631989f5/milestones/5c07d0768b83251b841a6a6d

But it never showed up. Its just in limbo...

<img width="1198" alt="screen shot 2018-12-10 at 1 41 09 am" src="https://user-images.githubusercontent.com/14945613/49705184-0dc3ff80-fc1d-11e8-874b-a92557bc9a0a.png">

Help! :-D

**Comment by satyavh** [comment]
This is not an issue caused by token integration, it's caused by failed tx and likely incomplete logic handling failed txs. As such is not part of the Dai reward. 

But I do feel your pain and dived into it.

Question:
You delegated 28.52969300 ETH. Was that the full amount of a single donation?

> @dev
campaign ownerId: 95 / donation {_id: new ObjectId("5c030c4e6812a05976ba3cf2")}

**Comment by satyavh** [comment]
And this donation still shows up on your delegations page with 0.0000000ETH and a spinner?

The donation doesn't seem delegated at all, it is actually showing up here:
https://beta.giveth.io/campaigns/5b51bdbdf8ba4732631989f5

The only thing is that the UI is showing the spinner is incorrectly on the campaign.

**Comment by vojtechsimetka** [comment]
This will probably need reprocessing again

**Comment by GriffGreen** [comment]
@vojtechsimetka @satyavh any updates on this issue?

**Comment by vojtechsimetka** [comment]
I can try and reprocess tonight when everyone is asleep.

**Comment by GriffGreen** [comment]
awesome! Thx V :-D

**Comment by satyavh** [comment]
If you don't have a txHash than there was no delegation. 
There's no evidence at all in the db of a delegation, I doubt there will be something to reprocess

So @GriffGreen, what was the txHash?
And 
> does this donation still shows up on your delegations page with 0.0000000ETH and a spinner?

**Comment by GriffGreen** [comment]
the plot thickens:

<img width="879" alt="screen shot 2018-12-18 at 9 48 03 am" src="https://user-images.githubusercontent.com/14945613/50142322-229e3400-02aa-11e9-86a3-cc80427ac678.png">
In my delegations screen ^^^^ the donation is showing up as 20.whatever eth

<img width="1000" alt="screen shot 2018-12-18 at 9 47 36 am" src="https://user-images.githubusercontent.com/14945613/50142324-229e3400-02aa-11e9-8620-09698cc2f50a.png">
but the campaign says 28.whatever ETH 

And when i tired to donate the transaction came out as undefined

**Comment by satyavh** [comment]
This is possible. On the campaign we show what has been donated.
In the delegation tab we show what is left of each donation.

**Comment by satyavh** [comment]
The 7.69ETH of that donation was delegated to this milestone
https://beta.giveth.io/campaigns/5b51bdbdf8ba4732631989f5/milestones/5bf6a7afaad06c0b7db0191e

**Comment by satyavh** [comment]
I doesn't seem like your donation is still stuck in limbo :-)

**Comment by GriffGreen** [comment]
nice! that answers that one,

But I still can't delegate it... the error that pops up is:


      Description of the Error:
      Thare was a problem with the delegation transaction.

      Transaction link:
      https://rinkeby.etherscan.io/tx/undefined

**Comment by satyavh** [comment]
Is that all you get, anything in the console? 
Did your metamask popup at all? Are you signed in?

**Comment by satyavh** [comment]
And to which campaign/milestone did you try to delegate and what amount?

**Comment by GriffGreen** [comment]
Everything in this issue is the tennagraph campaign... trying to load the same donation to the same milestone: https://beta.giveth.io/campaigns/5b51bdbdf8ba4732631989f5/milestones/5c07d0768b83251b841a6a6d

<img width="1213" alt="screen shot 2018-12-19 at 5 58 14 am" src="https://user-images.githubusercontent.com/14945613/50199786-1fab4e00-0353-11e9-9085-6a21a47ffd4f.png">

<img width="1273" alt="screen shot 2018-12-19 at 5 59 07 am" src="https://user-images.githubusercontent.com/14945613/50199884-a3653a80-0353-11e9-8aac-d81beff62997.png">

**Comment by GriffGreen** [comment]
The plot thickens even more! There is something wrong with this campaign somewhere in feathers for sure... 

i went to edit a milestone and the address it took me to was: https://beta.giveth.io/campaigns/undefined/milestones/5c1b77536c76245cd3f97c8a/edit

<img width="1137" alt="screen shot 2018-12-22 at 3 51 12 am" src="https://user-images.githubusercontent.com/14945613/50369902-19a8ad80-059d-11e9-8553-e014f6cae35e.png">

Also this is the campaign that i have to keep renaming... see issue #490 maybe something in that fix is causing this error.

**Comment by GriffGreen** [comment]
Ok now this is just wild...

I donated 28.5 ETH to this milestone:
https://beta.giveth.io/campaigns/5b51bdbdf8ba4732631989f5/milestones/5c07d0768b83251b841a6a6d
In this transaction:
https://etherscan.io/tx/0xfda47f04551b6b34564e75e67b7236907a2924403ea304f7abc95dd999474558

and the dapp displays that it only has an amount donated of 14.160307 ETH
..
Even tho it displays 28.5 ETH was donated!

<img width="922" alt="screen shot 2018-12-24 at 9 54 16 pm" src="https://user-images.githubusercontent.com/14945613/50406554-3294a700-07c7-11e9-8319-10cf5be60af0.png">


This is a huge thing.. Also I have a bunch of other delegations and donations missing in the dapp :-( I assume its in the smart contracts so i'm not sooooooo worried... but now i wonder if they are in the milestones.... and yeah the accounting is broken in the UI for sure

**Comment by GriffGreen** [comment]
<img width="1235" alt="screen shot 2018-12-26 at 2 33 38 pm" src="https://user-images.githubusercontent.com/14945613/50447419-4cdf9980-091b-11e9-9953-8310cd7eec01.png">
Welp shit :-(
Glad that milestone is gone.... but now what do we do, I donated 28.5 ETH to the milestone, it got 14.something... and we paid out the milestone so now the rest of the original donation is just gone. :-(

**Comment by GriffGreen** [comment]
Ok! I think I found it!

Its the 14.05 in the delegation page for my 83939 address
<img width="1065" alt="screen shot 2018-12-27 at 9 40 45 am" src="https://user-images.githubusercontent.com/14945613/50476044-e57f2380-09c6-11e9-8ba9-d20d81705633.png">

**Comment by GriffGreen** [comment]
Ok so the big thing that needs to be fixed here is that the 20.8 ETH Donation that Aragon made is stuck... it cant be delegated anywhere but it was able to be delegate partially before the upgrade... This is likely a bug that was created during the last upgrade, which is why the DAI reward tag is on it. You can't break the dapp with a feature and then call the feature done.

Ideally a post mortem should be written here to determine if there is further action that needs to be done with this campaign. If this post mortem proves that the last upgrade didn't cause the bug, or that it did but it has been fixed and a course of action to get the funds back can be proposed, then the DAI reward label can be removed.

This post mortem would address the locked aragon donation and what happened when i donated 28.5 ETH and only 14 went into the milestone? Is this Campaign cursed? Should we call an exorcist? Maybe just cancel it and make a new one? 

What are we going to do?

**Comment by perissology** [comment]
okay, that 20.8 ETH is actually in the milestone as it is supposed to be.

For some reason (I think maybe when Vojtech reprocessed events), there are events missing from the database. The tx where this delegation occurred is missing, thus why feathers isn't correct.

I'm re-processing the events already b/c a similar issue happened elsewhere, so the delegations should appear correct once that is done.

It looks like it was a UI or out-of-sync bug that allowed you to donate the 28.5 ETH. According to the contracts, 7.69 ETH was donated to 'Max Work 20.10 – 20.11' milestone [here](https://rinkeby.etherscan.io/tx/0xf9e0f9ecf60dd2705b49603e8a7bc73fc56881b297d48deacac8a34bbf7953ef)

That left only 20.839693 ETH (the broken delegation) to be delegated. This was delegated to 'Bohdan 12.11 – 30.11' milestone  [here](https://rinkeby.etherscan.io/tx/0xb129e56f8f81f9380408e83682fab2959ebd34ee6b3d332a8b139712ba20cf33)

I can't say this was directly due to the DAI integration and was most likely due to a reprocessing sna-foo. I did make a [fix](ba356f95fd09d3d43c2ed2cdeb84764193285f8e) that allows feathers to sync from scratch again. I know Vojtech was not able to sync from scratch recently, which may have lead to these issues.

Leaving this open until we are synced again and the milestone is paid out correctly

**Comment by perissology** [comment]
okay, can you see if that milestone is able to disburse? You should be able to.

ignore the total donated, I'll fix those tomorrow

**Comment by perissology** [comment]
All balances should now be correct in the dapp.

**Comment by GriffGreen** [comment]
Done!
